CMake/Dependencies/libkvsCommonLws-CMakeLists.txt

```diff
@@ -6,10 +6,12 @@ include(ExternalProject)
 
 ExternalProject_Add(libkvsCommonLws-download
     GIT_REPOSITORY    https://github.com/awslabs/amazon-kinesis-video-streams-producer-c.git
-    GIT_TAG           v1.5.2
+    GIT_TAG           link-mbedtls-properly
```

Comment:
```diff
target_link_libraries(kvsCommonLws
+           ${PRODUCER_CRYPTO_LIBRARY}
            ${OPENSSL_CRYPTO_LIBRARY}
            ${OPENSSL_SSL_LIBRARY}
            ${LIBWEBSOCKETS_LIBRARIES}
```

Looking at the cmake 1-line change in this Producer C branch, its seems we should only link the `${OPENSSL_CRYPTO_LIBRARY} ${OPENSSL_SSL_LIBRARY}` libraries if not opting for MbedTLS - or better yet only have the `PRODUCER_CRYPTO_LIBRARY` one which should cover both cases:

```
set(PRODUCER_CRYPTO_LIBRARY
        OpenSSL::Crypto
        OpenSSL::SSL)
if (USE_MBEDTLS)
    set(CPRODUCER_COMMON_TLS_OPTION KVS_USE_MBEDTLS)
    set(PRODUCER_CRYPTO_LIBRARY
            MbedTLS
            MbedCrypto)
endif()
```

So I propose the following change:

```diff
target_link_libraries(kvsCommonLws
            ${PRODUCER_CRYPTO_LIBRARY}
-           ${OPENSSL_CRYPTO_LIBRARY}
-           ${OPENSSL_SSL_LIBRARY}
            ${LIBWEBSOCKETS_LIBRARIES}
```
And with a nit variable name change that makes more sense as there are multiple libraries and to be more aligned with how we also named `LIBWEBSOCKETS_LIBRARIES`:

```diff
target_link_libraries(kvsCommonLws
-           ${PRODUCER_CRYPTO_LIBRARY}
+           ${PRODUCER_CRYPTO_LIBRARIES}
            ${LIBWEBSOCKETS_LIBRARIES}
```

Reply:
Yes I thought of this as well, I have no idea why it was set up the way it is, however the openssl cmake variables will be blank in this case so it will be ignored by the target link libraries (I verified this).  Preference though would be to do as you suggested as the final change.